core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -466,4 +468,13 @@ private static Pair<Region, Optional<Boolean>> transformFipsPseudoRegionIfNecess
 
         return Pair.of(region, Optional.empty());
     }
+
+    private Supplier<ProfileFile> profileFileSupplierOrFallbackToProfileFile(SdkClientConfiguration config) {
+        Supplier<ProfileFile> profileFile = config.option(SdkClientOption.PROFILE_FILE_SUPPLIER);
+        if (Objects.nonNull(profileFile)) {
```

Comment:
Comment: The code base mostly uses `==` for null testing, so this is a slight deviation from the standard. Up to you though.

Reply:
Since I used == and != in the majority of other checks, i've decided to change this one.